acl_loader/main.py

```diff
@@ -453,6 +481,41 @@ def convert_ip(self, table_name, rule_idx, rule):
 
         return rule_props
 
+    def convert_icmp(self, table_name, rule_idx, rule):
+        rule_props = {}
+
+        is_table_v6 = self.is_table_ipv6(table_name)
+        type_key = "ICMPV6_TYPE" if is_table_v6 else "ICMP_TYPE"
+        code_key = "ICMPV6_CODE" if is_table_v6 else "ICMP_CODE"
+
+        if rule.icmp.config.type or rule.icmp.config.type == 0:
```

Comment:
>rule.icmp.config.type [](start = 11, length = 21)

Is it possible that rule.icmp.config.type == "null"?

Reply:
"null" is a string so it will give an error when we try to cast it. The user shouldn't ever be passing null explicitly, we just have the value defined in the yang model so that the user is able to pass 0 as a valid ICMP type/code.

Reply:
Understood. To make code more readable, you may check `rule.icmp.config.type == "null"` first, if not equal, `rule.icmp.config.type` must be an integer, no need to cast.

---
In reply to: [586069013](https://github.com/Azure/sonic-utilities/pull/1469#discussion_r586069013) [](ancestors = 586069013)

Reply:
OK, that sounds good to me.